src/silo/preprocessing/preprocessing_database.cpp

```diff
@@ -109,22 +113,55 @@ preprocessing::Partitions PreprocessingDatabase::getPartitionDescriptor() {
    return preprocessing::Partitions(partitions);
 }
 
-ZstdFastaTable PreprocessingDatabase::generateSequenceTableFromFasta(
+ZstdTable PreprocessingDatabase::generateSequenceTableViaFile(
+   const std::string& table_name,
+   const std::string& reference_sequence,
+   const std::filesystem::path& file_path
+) {
+   const auto file_stem = file_path.stem().string();
+   for (const auto& entry : std::filesystem::directory_iterator(file_path.parent_path())) {
+      const auto entry_file_name = entry.path().filename().string();
+      if (!entry.is_regular_file() || !entry_file_name.starts_with(file_stem)) {
+         continue;
+      }
+      auto extensions = splitBy(entry_file_name, ".");
```

Comment:
Because we're processing multiple extensions in this case and .extension only returns the last one I think

Reply:
But can't we iterate on `file_path.extension`?
* Checking it.
  * if we find fasta or sam, done.
  * if we find a compression extension, memorize it, pop the extension and reiterate.

It's basically what you already do, but a bit more expressive. But it's a nitpick, you can also leave it as it is. :+1:

Reply:
I'm not sure what you mean. Doing a foreach on entry.path().extension() doesn't really work it just does one iteration with the last extension (so if the file is something.fasta.zst it will only get the .zst, it's not an iterator it just gets the .zst). In order to get it to work it would have to make temp strings with the last extension cut off and call .extension() again (which is how I did it initially but realized it was very inelegant)

Reply:
Ok, then let's leave it as it is.